src/StoredEvents/Repositories/EloquentStoredEventRepository.php

```diff
@@ -91,8 +91,9 @@ public function persistMany(array $events, string $uuid = null, int $aggregateVe
     {
         $storedEvents = [];
 
+        /** @var \Spatie\EventSourcing\StoredEvents\ShouldBeStored $event */
         foreach ($events as $event) {
-            $storedEvents[] = $this->persist($event, $uuid, $aggregateVersion);
+            $storedEvents[] = $this->persist($event, $uuid, $event->aggregateRootVersion() ?? $aggregateVersion);
```

Comment:
I'm keeping this `?? $aggregateVersion` as a backwards compatible fallback, but we might further refactor it in the future.